src/transport/UnauthenticatedSessionTable.h

```diff
@@ -0,0 +1,204 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/core/CHIPError.h>
+#include <lib/core/ReferenceCounted.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/Pool.h>
+#include <lib/support/ReferenceCountedHandle.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <system/TimeSource.h>
+#include <transport/raw/PeerAddress.h>
+
+namespace chip {
+namespace Transport {
+
+class UnauthenticatedSession;
+using UnauthenticatedSessionHandle = ReferenceCountedHandle<UnauthenticatedSession>;
+
+class UnauthenticatedSessionDeleter
+{
+public:
+    // This is a no-op because life-cycle of UnauthenticatedSessionTable is rotated by LRU
+    static void Release(UnauthenticatedSession * entry) {}
+};
+
+/**
+ * @brief
+ *   An UnauthenticatedSession stores the binding of TransportAddress, and message counters.
+ *
+ *   The entries are rotated using LRU, but entry can be hold by using UnauthenticatedSessionHandle, which increase the reference
```

Comment:
The LRU comment should be part of the 'table' since it is a propery of the table not of the session object.